tests/unit/plugins/modules/test_lambda_layer.py

```diff
@@ -0,0 +1,500 @@
+            "description": "lambda layer first version",
+            "layer_version_arn": "arn:aws:lambda:eu-west-2:123456789012:layer:testlayer:1",
+            "version": 1
+        }
+    ]
+    params = {"name": "testlayer", "version": 1}
+    with pytest.raises(lambda_layer.LambdaLayerFailure):
+        lambda_layer.delete_layer_version(lambda_client, params)
+
+
+@pytest.mark.parametrize(
+    "b_s3content",
+    [
+        (True),
+        (False)
+    ]
+)
+@patch(mod_list_layer)
+def test_create_layer(m_list_layer, b_s3content):
+    params = {
+        "name": "testlayer",
+        "description": "ansible units testing sample layer",
+        "content": {},
+        "license_info": "MIT"
+    }
+
+    lambda_client = MagicMock()
+
+    lambda_client.publish_layer_version.return_value = {
+        'CompatibleRuntimes': [
+            'python3.6',
+            'python3.7',
+        ],
+        'Content': {
+            'CodeSha256': 'tv9jJO+rPbXUUXuRKi7CwHzKtLDkDRJLB3cC3Z/ouXo=',
+            'CodeSize': 169,
+            'Location': 'https://awslambda-us-west-2-layers.s3.us-west-2.amazonaws.com/snapshots/123456789012/my-layer-4aaa2fbb',
+        },
+        'CreatedDate': '2018-11-14T23:03:52.894+0000',
+        'Description': "ansible units testing sample layer",
+        'LayerArn': 'arn:aws:lambda:us-west-2:123456789012:layer:my-layer',
+        'LayerVersionArn': 'arn:aws:lambda:us-west-2:123456789012:layer:testlayer:1',
+        'LicenseInfo': 'MIT',
+        'Version': 1,
+        'ResponseMetadata': {
+            'http_header': 'true',
+        },
+    }
+
+    expected = {
+        "changed": True,
+        "layer_versions": [
+            {
+                'compatible_runtimes': ['python3.6', 'python3.7'],
+                'content': {
+                    'code_sha256': 'tv9jJO+rPbXUUXuRKi7CwHzKtLDkDRJLB3cC3Z/ouXo=',
+                    'code_size': 169,
+                    'location': 'https://awslambda-us-west-2-layers.s3.us-west-2.amazonaws.com/snapshots/123456789012/my-layer-4aaa2fbb'
+                },
+                'created_date': '2018-11-14T23:03:52.894+0000',
+                'description': 'ansible units testing sample layer',
+                'layer_arn': 'arn:aws:lambda:us-west-2:123456789012:layer:my-layer',
+                'layer_version_arn': 'arn:aws:lambda:us-west-2:123456789012:layer:testlayer:1',
+                'license_info': 'MIT',
+                'version': 1
+            }
+        ]
+    }
+
+    zfile_handler = None
+
+    if b_s3content:
+        params["content"] = {
+            "s3_bucket": "mybucket",
+            "s3_key": "mybucket-key",
+            "s3_object_version": "v1"
+        }
+        content_arg = {
+            "S3Bucket": "mybucket",
+            "S3Key": "mybucket-key",
+            "S3ObjectVersion": "v1"
+        }
+    else:
+        binary_data = b"simple lambda layer content"
+        zfile_handler = NamedTemporaryFile(delete=False)
```

Comment:
You can use `tmp_path` fixture to be sure your temp file is clean up at the end of the test. https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html